clusterctl/main_integration_test.go

```diff
@@ -0,0 +1,174 @@
+// +build integration
+
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main_test
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"syscall"
+	"testing"
+)
+
+const (
+	clusterctlBinaryName = "clusterctl"
+	workingDirectoryName = "clusterctl"
+)
+
+// run these tests with the flag "-update" to update the values stored in all of the golden files
+var update = flag.Bool("update", false, "update .golden files")
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	err := setupPrerequisites()
+	if err != nil {
+		fmt.Printf("unable to setup prerequisites: %v\n", err)
```

Comment:
Generally we are using glog for proper log statements for additional details like proc, file, line number, timestamp etc.

Reply:
Agree that glog gives us that information for free and it is useful. I think it is most useful for long running services. Glog itself brings some issues around not playing nice with flags and init(...) methods. I went with Printf because this is a test program and not a long running application.